### PR TITLE
[librsvg] Add pkgconfig file

### LINKED ISSUE #31876: [librsvg] pkgconfig file is missing
### Is your feature request related to a problem? Please describe.

When installing librsvg, there is no pkg-config file installed, which results in a failure, at configure time, when that file is requested from other pkg-dependencies.

### Proposed solution

add that config file

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Cheney-W**
Fixes https://github.com/microsoft/vcpkg/issues/31876

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Below usage test passed with x64-windows triplet:
```
find_package(PkgConfig REQUIRED)
pkg_check_modules(librsvg REQUIRED IMPORTED_TARGET librsvg)
target_link_libraries(main PkgConfig::librsvg)
```

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/librsvg/librsvg.pc.in [hunk 1]
```diff
@@ -0,0 +1,12 @@
+prefix=${pcfiledir}/../..
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: librsvg
+Description: library that renders svg files
+Version: @LIBRSVG_VERSION@
```
Comment (dg0yt) [inline ports/librsvg/librsvg.pc.in:8]: Why not use `@VERSION@` here?

File: ports/librsvg/librsvg.pc.in [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Requires: glib-2.0 gobject-2.0 gdk-pixbuf-2.0 cairo pango pangocairo
+Requires.private:
+Libs: -L${libdir} -lrsvg-2.40 -lpixbufloader-svg
```
Comment (dg0yt) [inline ports/librsvg/librsvg.pc.in:11]: I don't think this is the optimal use of `Requires[.private]` and `Libs[.private]`.
In contrast to upstream it also lacks `-lm` (for non-MSVC).

File: ports/librsvg/usage [hunk 3]
```diff
@@ -0,0 +1,10 @@
+librsvg can be imported via CMake FindPkgConfig module:
+
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(librsvg REQUIRED IMPORTED_TARGET librsvg)
+    target_link_libraries(main PkgConfig::librsvg)
+
+vcpkg provides proprietary CMake targets:
+
+    find_package(unofficial-librsvg CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE unofficial::librsvg::rsvg-2.40 unofficial::librsvg::pixbufloader-svg)
```
Comment (dg0yt) [inline ports/librsvg/usage:10]: Does it make sense to recommend a target name with ABI version?
Do users always need to link to `unofficial::librsvg::pixbufloader-svg`?

**Review by dg0yt — COMMENTED** [review]
Why not use upstream's https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.40.20/librsvg.pc.in directly?

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/librsvg/librsvg.pc.in (hunk 2 above)
Comment (Cheney-W) [reply ports/librsvg/librsvg.pc.in]: Thanks for your comment.

File: ports/librsvg/librsvg.pc.in (hunk 1 above)
Comment (Cheney-W) [reply ports/librsvg/librsvg.pc.in:8]: @VERSION@ does not exist in CMakeLists.txt.

File: ports/librsvg/usage (hunk 3 above)
Comment (Cheney-W) [reply ports/librsvg/usage]: I have removed the ABI version.
> Do users always need to link to unofficial::librsvg::pixbufloader-svg?

I'm not sure, so I keep the usage same as before.

**Comment by Cheney-W** [comment]
> Why not use upstream's https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.40.20/librsvg.pc.in directly?

The variables in the upstream `librsvg.pc.in `file cannot be recognized.

**Comment by dg0yt** [comment]
Even if the port uses a vendored build system, shouldn't it keep upstream's chosen names for binaries?
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-rename-binaries-outside-the-names-given-by-upstream

**Comment by dg0yt** [comment]
> The variables in the upstream `librsvg.pc.in` file cannot be recognized.

Normally this is resolved by just setting them...

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
> shouldn't it keep upstream's chosen names for binaries? https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-rename-binaries-outside-the-names-given-by-upstream

I agree with @dg0yt here, why is renaming the targets necessary?

**Comment by Cheney-W** [comment]
> why is renaming the targets necessary?

According to the contents of the https://gitlab.gnome.org/GNOME/librsvg/-/blob/main/librsvg.pc.in
    `Libs: -L${libdir} -lrsvg-@RSVG_API_MAJOR_VERSION@ -lm`

**Comment by BillyONeal** [comment]
I see, this is our external build system, not a patch. OK

**Comment by Febbe** [comment]
librsvg contains now `Libs: "-L${libdir}" -lrsvg-2 -lm` on Windows, which must not happen in any pc file on Windows. Maybe this should be added in `vcpkg_fixup_pkgconfig()` which will fix it for every pc file.

**Comment by BillyONeal** [comment]
> Maybe `-lm` should be removed via `vcpkg_fixup_pkgconfig()` which will fix it for every pc file.

We talked about this today and are a bit to nervous to open versioning cans of worms trying to touch an existing thing that would affect port versions that already exist. And that doesn't mean there will never be a library named `m.lib` someone might want to use on Windows. In cases where we know this was the intent, one could imagine patching upstream to fix it or using tools like `vcpkg_replace_string` to deal with it, though the latter might be dangerous hammer with short strings like `" -lm"` since those can be in directory names or similar too.

**Pull request merged**